pages/components/docs/core-concepts

```diff
@@ -0,0 +1,71 @@
+# Core Concepts
+
+## The `as` prop
+The `as` prop is a feature that all of our components get from [styled-components](https://www.styled-components.com). It allows you to pass a HTML tag or another component to a Primer Component to be rendered as the base tag of that component.
+
+
+For example, say you are using a `Button` component, and you really need to apply `Flex` styles to it. You can compose `Flex` and `Button` like so:
+
+```
+<Button as={Flex}/>
```

Comment:
I tried it in a CodeSandbox and it ends up rendering a `<div>` with button styles instead of a `<button>` element: 

![image](https://user-images.githubusercontent.com/4608155/63197216-0e265d00-c02c-11e9-94a3-1934f2c0743a.png)
 https://codesandbox.io/s/primer-components-template-uzhlz
 

I'm not sure if we should be recommending people use the `as` prop with other Primer components like this since the rendered output can sometimes be unexpected. Maybe a good rule of thumb could be: only use the `as` prop if you can't achieve the same result by nesting components. For example:

```
<Button>
  <Flex></Flex>
</Button>
```
is more predictable than:
```
<Button as={Flex}>
```

Reply:
Yeah I agree 🤔 I don't love the `as` prop because of this, but it is something that tends to get used a lot. Maybe we should even go as far as to discourage it's use cc @broccolini